File: NendVideoMoPubCustomEventSample/NendVideoMoPubCustomEventSample/ViewController.m
//
//  ViewController.m
//  NendVideoMoPubCustomEventSample
//
//  Copyright © 2017年 F@N Communications. All rights reserved.
//

#import "ViewController.h"

#import "MoPub.h"
#import "MPRewardedVideo.h"
#import "NendInstanceMediationSettings.h"

static NSString *const kAdUnitId = @"your ad unit id";

@interface ViewController () <MPRewardedVideoDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [[MoPub sharedInstance] initializeRewardedVideoWithGlobalMediationSettings:nil delegate:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)load:(id)sender {
    NendInstanceMediationSettings *settings = [NendInstanceMediationSettings new];
    settings.userId = @"your user id";
    [MPRewardedVideo loadRewardedVideoAdWithAdUnitID:kAdUnitId withMediationSettings:@[settings]];
}

- (IBAction)show:(id)sender {
    if ([MPRewardedVideo hasAdAvailableForAdUnitID:kAdUnitId]) {
        [MPRewardedVideo presentRewardedVideoAdForAdUnitID:kAdUnitId fromViewController:self withReward:nil];
    }
}


#pragma mark - MPRewardedVideoDelegate

- (void)rewardedVideoAdDidLoadForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdDidFailToLoadForAdUnitID:(NSString *)adUnitID error:(NSError *)error {
    NSLog(@"%s: %@, error: %@", __FUNCTION__, adUnitID, error);
}
- (void)rewardedVideoAdDidFailToPlayForAdUnitID:(NSString *)adUnitID error:(NSError *)error {
    NSLog(@"%s: %@, error: %@", __FUNCTION__, adUnitID, error);
}
- (void)rewardedVideoAdWillAppearForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdDidAppearForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdWillDisappearForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdDidDisappearForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdWillLeaveApplicationForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdDidReceiveTapEventForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s: %@", __FUNCTION__, adUnitID);
}
- (void)rewardedVideoAdShouldRewardForAdUnitID:(NSString *)adUnitID reward:(MPRewardedVideoReward *)reward {
    NSLog(@"%s: %@, reward: %@ %@", __FUNCTION__, adUnitID, reward.currencyType, reward.amount);
}

@end

File: NendVideoMoPubCustomEventSample/NendVideoMoPubCustomEventSample_Swift/ViewController.swift
//
//  ViewController.swift
//  NendVideoMoPubCustomEventSample_Swift
//
//  Copyright © 2017年 F@N Communications. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let adUnitId = "your ad unit id"

    override func viewDidLoad() {
        super.viewDidLoad()
        MoPub.sharedInstance().initializeRewardedVideo(withGlobalMediationSettings: [], delegate: self)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func loadAd(_ sender: Any) {
        let settings = NendInstanceMediationSettings()
        settings.userId = "your user id"
        MPRewardedVideo.loadAd(withAdUnitID: adUnitId, withMediationSettings: [settings])
    }
    
    @IBAction func showAd(_ sender: Any) {
        if MPRewardedVideo.hasAdAvailable(forAdUnitID: adUnitId) {
            MPRewardedVideo.presentAd(forAdUnitID: adUnitId, from: self, with: nil)
        }
    }
}

extension UIViewController: MPRewardedVideoDelegate {
    public func rewardedVideoAdDidLoad(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdDidFailToLoad(forAdUnitID adUnitID: String!, error: Error!) {
        print("\(#function) error: \(error)")
    }
    
    public func rewardedVideoAdDidFailToPlay(forAdUnitID adUnitID: String!, error: Error!) {
        print("\(#function) error: \(error)")
    }
    
    public func rewardedVideoAdWillAppear(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdDidAppear(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdWillDisappear(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdDidDisappear(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdWillLeaveApplication(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdDidReceiveTapEvent(forAdUnitID adUnitID: String!) {
        print("\(#function) \(adUnitID)")
    }
    
    public func rewardedVideoAdShouldReward(forAdUnitID adUnitID: String!, reward: MPRewardedVideoReward!) {
        print("\(#function) \(reward.currencyType) : \(reward.amount)")
    }
}
